Add /review-pr skill and wire it into the Claude Code Review workflow (#4351)
cowprotocol/services: docs/review-context/database-migrations.md

# Review Context — Database Migrations

Loaded by `COW_PR_REVIEW_SKILL.md §3` when a PR touches `database/sql/**`.

This file extends — does not replace — the reminder in `.github/nitpicks.yml`. The nitpick bot posts the generic DB warning automatically; the reviewer applies judgment using the checks below.

## Non-negotiables (all **High** until addressed)

1. **Reversibility.** The migration must state whether it's reversible. If yes, the rollback script is present or explicitly linked in the PR description. If no, the PR explains *why* irreversibility is acceptable for this change. Missing either → **High**.

> 2. **Compatibility window.** During rollout, k8s starts the new autopilot, runs Flyway, and only then shuts down the old pod. That overlap means the **previous** version must still be able to process requests on the **new** schema. Any migration that breaks the old schema path (e.g. drops a column the old code reads) → **High** until the PR lays out an n-1 compatibility plan, typically:

Reviewer: I think being specific for the DB setup makes a bit more sense as the way how our migrations get rolled out affect how they need to be written.
It also makes sense to point out that we have a staging and production environment that are independent and get rolled out at different times (this can break the shadow environment for example - not just related to the DB).

Author: Kept this, and rewrote it. Opening now explains why the file is CoW-specific (checks dictated by rollout, not by SQL). 
The new "Rollout shape" section spells out both points explicitly: k8s pod overlap (n-1 against new schema) and the independent staging/prod cadences that can break shadow. Added a question template asking how the change behaves on shadow during the staging→production gap.

## Non-negotiables (default **High** until addressed)

1. **Reversibility.** State whether the migration is reversible. If yes, include or link the rollback script. If no, the PR explains *why* irreversibility is acceptable. Missing either → **High**.
2. **n-1 schema compatibility.** The previous app version must still function against the new schema. A migration that drops or renames a column, narrows a type, or adds a `NOT NULL` constraint without code already tolerating both shapes → **High** until the rollout plan is spelled out (typically: ship change in three releases — add → migrate code → drop).
3. **Blocking index creation on hot tables.** Use `CREATE INDEX CONCURRENTLY` on anything in the auction/settlement critical path (`orders`, `trades`, `auctions`, `settlements`, `order_events`, `auction_orders`, `quotes`, `order_quotes`). A blocking `CREATE INDEX` on one of these → **High**.
4. **Authoritative table list.** New tables must appear in `crates/database/src/lib.rs` (search for the top-level table list). Missing → **Medium** (CI may also catch it; still the author's responsibility).

> 5. **README.** Schema changes update the SQL or DB README, whichever is the convention at review time. Missing → **Medium**.

Author: The reason is a bit vague, but this is a genuine contradiction. fixed this

# Review Context — Database Migrations

Loaded by `COW_PR_REVIEW_SKILL.md §3` when a PR touches `database/sql/**`.

This file extends — does not replace — the reminder in `.github/nitpicks.yml`. The nitpick bot posts the generic warning automatically; the reviewer applies judgment using the checks below.

The checks are dictated by *how* we roll out — not by SQL itself.

## Rollout shape

Two facts shape every check below:

1. **k8s rolls pods, not the cluster.** During a release, the new autopilot starts and runs Flyway *before* the old pod is shut down. For a non-trivial overlap window the previous version is processing requests against the new schema. Anything that breaks that path breaks production.

> 2. **Staging and production are independent and roll out at different times.** A migration lands in staging first; production follows on its own cadence. While the gap is open, our shadow environment can be running an *older* code version against the *newer* schema (or vice versa, briefly). Any change that breaks that combination breaks shadow without breaking the obvious path the author tested.

Reviewer 2: do we still have a shadow environment?

3. **Blocking index creation on hot tables.** Use `CREATE INDEX CONCURRENTLY` on anything in the auction/settlement critical path (`orders`, `trades`, `auctions`, `settlements`, `order_events`, `auction_orders`, `quotes`, `order_quotes`). A blocking `CREATE INDEX` on one of these → **High**.
4. **Authoritative table list.** New tables must appear in `crates/database/src/lib.rs` (search for the top-level table list). Missing → **Medium** (CI may also catch it; still the author's responsibility).
5. **README.** Schema changes update the SQL or DB README, whichever is the convention at review time. Missing → **Medium**.

## Other shapes that usually warrant **High**

- `ALTER TABLE ... ADD COLUMN NOT NULL` on a multi-million-row table without a default — table lock plus slow backfill. Remedy: add nullable, batched backfill, then `NOT NULL` in a later migration.
- Renaming a column in a single migration rather than the three-release add → migrate → drop pattern.

> - Adding a `UNIQUE` constraint without first verifying current data is unique (the migration will fail in prod if duplicates exist).

Reviewer 3: ```suggestion
- Adding a `UNIQUE` constraint without first verifying current data is unique (migrations fail when duplicates exist).
```

> - Type changes in place on a large table without a plan for the rewrite cost.

Reviewer 3: Not sure what "rewrite cost" here means; but if it's code, I don't even expect this to happen without the required Rust updates because E2E tests *should* fail

Author: rewrote this part

## Usually worth flagging as **Medium**

> - Migration scripts without comments when the *what* is non-obvious.

Reviewer 3: This is not very clear and I'd expect this to lead to extra review notes
